### PR TITLE
SOV-3117: reward claiming

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.atlassian.net/browse/SOV-2799
https://sovryn.atlassian.net/browse/SOV-3117

### CODE REVIEW

File: apps/frontend/src/app/5_pages/RewardsPage/components/Staking/Staking.tsx [hunk 1]
```diff
@@ -29,29 +28,34 @@ export const Staking: FC = () => {
...
+      (!earnedFees.some(earnedFee => decimalic(earnedFee.value).gt(0)) &&
+        !decimalic(liquidSovClaimAmount).gt(0)) ||
+      !account,
+    [account, earnedFees, liquidSovClaimAmount],
+  );
 
-    return [
-      ...earnedFees.map(earnedFee => ({
-        type: getStakingRevenueType(earnedFee.token),
+  const rows1 = useMemo(
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/RewardsPage/components/Staking/Staking.tsx]:
```suggestion
  const rows = useMemo(
```

File: apps/frontend/src/app/5_pages/RewardsPage/components/Staking/components/WithdrawAllFees/WithdrawAllFees.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      onClick={onSubmit}
+      disabled={isClaimDisabled || loading}
+      loading={loading}
+      className="w-full lg:w-auto whitespace-nowrap"
+      dataAttribute="rewards-withdraw"
+    />
+  );
+};
+
+type UserCheckpoint = {
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/RewardsPage/components/Staking/components/WithdrawAllFees/WithdrawAllFees.tsx]: You could extract it to `WithdrawAllFees.types.ts`.

**Review by tiltom — CHANGES_REQUESTED** [review]
@creed-victor I tried the `Withdraw all` functionality but `SOV` and `ZUSD` fees were not withdrawn, even though the tx was successful. Can you look into it? My testing address is `0xaaa5a190accbc50f4f9c130b5876521e4d5f9d6c` if needed.

 Tx: https://explorer.testnet.rsk.co/tx/0xe5f87e3d3ba41131bee08e14cee47660a69b05bc1dc9eeef35dd05b34635be08

image

**Comment by creed-victor** [comment]
> @creed-victor I tried the `Withdraw all` functionality but `SOV` and `ZUSD` fees were not withdrawn, even though the tx was successful. Can you look into it? My testing address is `0xaaa5a190accbc50f4f9c130b5876521e4d5f9d6c` if needed.

This is how smart contract function works and it's expected. If user has a lot of tokens and checkpoints, they will have to send multiple transactions until all tokens/checkpoints are withdrawn.

**Review by tiltom — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**